rclcpp/src/rclcpp/init_options.cpp

```diff
@@ -96,7 +106,42 @@ InitOptions::finalize_init_options()
 const rcl_init_options_t *
 InitOptions::get_rcl_init_options() const
 {
+  std::lock_guard<std::mutex> init_options_lock(init_options_mutex_);
```

Comment:
Taking the mutex here doesn't help.
The returned pointer can be invalidated any time, as the API signature is unsafe.

For the moment, I would just delete the lock guard.